aten/src/ATen/cuda/CUDAGeneratorImpl.cpp

```diff
@@ -332,6 +416,22 @@ DeviceType CUDAGeneratorImpl::device_type() {
   return DeviceType::CUDA;
 }
 
+/**
+ * Retrieves the current state of the CUDA generator as a constant reference.
+ * This function provides read-only access to the generator's state.
+ */
+CUDAGeneratorState& CUDAGeneratorImpl::state() {
+  TORCH_CHECK(
+      current_state_id_ < states_.size(), "The state index is invalid.");
+  return states_[current_state_id_];
+}
+
+const CUDAGeneratorState& CUDAGeneratorImpl::state() const {
+  TORCH_CHECK(
+      current_state_id_ < states_.size(), "The state index is invalid.");
+  return states_[current_state_id_];
+}
+
```

Comment:
The state() method provides access to the current state of the CUDA generator in use. The pointer to the device state is captured in the CUDA Graph, so it could capture the state-switch behavior.